torchtyping/tensor_details.py

```diff
@@ -97,6 +97,8 @@ def check(self, tensor: torch.Tensor) -> bool:
                 and self_name != tensor_name
             ):
                 return False
+            if isinstance(self_size, str):
+                return True
```

Comment:
This looks wrong, as this breaks out of the loop early. Only `return False` should appear in the loop, and it's only if all of those checks pass that we then hit the `return True` at the end of the loop.

Instead I think this should be merged into the next check: `if (self_size not in (-1, tensor_size)) or isinstance(self_size, str): return False`

Reply:
you're right; I missed the loop! (I tried hard to find a test-case that would fail before fixing this, but because some things are checked twice, I didn't actually manage to find one).  I fixed this and added a test that would "almost" fail without the fix. :)